src/main/java/dev/lukebemish/ziggradle/ZigTargetTriple.java

```diff
@@ -0,0 +1,95 @@
+package dev.lukebemish.ziggradle;
+
+import dev.lukebemish.ziggradle.internal.DefaultZigTargetTriple;
+import org.gradle.api.Named;
+import org.jetbrains.annotations.NotNull;
+
+import javax.annotation.Nullable;
+import java.io.Serializable;
+
+public interface ZigTargetTriple extends Named, Serializable {
```

Comment:
Why is this named? Shouldn't it be a nested object with a property for each piece?